File: python/save.js
const fs = require('fs');
const path = require('path');

const CID = 'Qma7YyzwcUpWHJh4ojC5r8LXTF1LamSzat6JHwYCAmGr4V';

async function retrieveDataFromCID(cid) {
    const url = `http://127.0.0.1:8080/ipfs/${cid}`;
    try {
        const fetch = (await import('node-fetch')).default; // Dynamically import node-fetch
        const response = await fetch(url);
        if (!response.ok) {
            throw new Error(`Network response was not ok. Status: ${response.status}`);
        }
        const data = await response.text();
        return data;
    } catch (error) {
        throw new Error(`Failed to retrieve data: ${error.message}`);
    }
}

retrieveDataFromCID(CID)
    .then(secret => {
        console.log('Secret:', secret);

        const downloadDir = path.join(__dirname, 'download');
        if (!fs.existsSync(downloadDir)) {
            fs.mkdirSync(downloadDir);
        }

        const downloadPath = path.join(downloadDir, 'data.json');
        fs.writeFileSync(downloadPath, JSON.stringify({ secret }, null, 2));
        console.log('Data saved to', downloadPath);
    })
    .catch(console.error);

File: python/save.py
import os
import requests

# example ipfs link: http://localhost:8080/ipfs/QmNNnmALiGdBgiYHPffoyDQr1L5LKQeG2cnSC39uztgq3D

CID = 'QmYddbqBYgM6LrKKQfpAhTpPzE8qaJW1CpHdkjP35kCqoZ'

def retrieve_data_from_cid(cid):
    url = f'http://localhost:8080/ipfs/{cid}'
    try:
        response = requests.get(url)
        response.raise_for_status()
        return response.text
    except requests.RequestException as error:
        raise Exception(f'Failed to retrieve data: {error}')

try:
    secret = retrieve_data_from_cid(CID)
    print('Secret:', secret)
    download_path = os.path.join(os.path.dirname(__file__), 'download', 'data.json')
    os.makedirs(os.path.dirname(download_path), exist_ok=True)
    with open(download_path, 'w') as file:
        file.write(secret)
    print('Data saved to', download_path)
except Exception as error:
    print(error)